Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_test.py

```diff
@@ -0,0 +1,541 @@
+import demistomock as demisto
+import json
+
+
+def mock_demisto(mocker):
+    mocker.patch.object(demisto, 'params', return_value={'proxy': True,
+                                                         'url': 'https://api.securitycenter.windows.com',
+                                                         'tenant_id': '1234',
+                                                         'enc_key': 'key',
+                                                         'auth_id': '1234567@1234567',
+                                                         'fetch_severity': 'Informational,Low,Medium,High',
+                                                         'fetch_status': 'New'})
+    mocker.patch.object(demisto, 'getLastRun', return_value={'last_alert_fetched_time': "2018-11-26T16:19:21.840980"})
+    mocker.patch.object(demisto, 'incidents')
+
+
+def atp_mocker(mocker):
+    import MicrosoftDefenderAdvancedThreatProtection as atp
+    with open('test_data/alerts.json', 'r') as f:
+        alerts = json.loads(f.read())
+    mocker.patch.object(atp, 'list_alerts_request', return_value=alerts)
+
+
+def test_fetch(mocker):
```

Comment:
by moving functionality in fetch to outer functions, you'll be able to thoroughly test each part of the functionality separately.
this ability is extremely important in fetch, which is very bug prone.
please add significantly more tests to it after you export the functions - test that the next run values are updated  correctly, that the limit works as expected,  test every assumption that you have about what works in this function